mycore-base/src/main/java/org/mycore/datamodel/common/MCRFileBaseCacheObjectIDGenerator.java

```diff
@@ -50,36 +54,40 @@ public class MCRFileBaseCacheObjectIDGenerator implements MCRObjectIDGenerator {
     static ConcurrentHashMap<String, ReentrantReadWriteLock> locks = new ConcurrentHashMap<>();
 
     private static Path getCacheFilePath(String baseId) {
+        Path idCachePath = getCacheDirPath();
 
-        Path dataDir = getDataDirPath();
-
-        Path idCachePath = dataDir.resolve("id_cache");
-        if (!Files.exists(idCachePath)) {
+        Path cacheFile = MCRUtils.safeResolve(idCachePath, baseId);
+        if (!Files.exists(cacheFile)) {
             synchronized (MCRFileBaseCacheObjectIDGenerator.class) {
-                if (!Files.exists(idCachePath)) {
+                if (!Files.exists(cacheFile)) {
                     try {
-                        Files.createDirectory(idCachePath);
+                        Files.createFile(cacheFile);
                     } catch (IOException e) {
-                        throw new MCRException(
-                            "Could not create " + idCachePath.toAbsolutePath() + " directory", e);
+                        throw new MCRException("Could not create " + cacheFile.toAbsolutePath(), e);
                     }
                 }
             }
         }
+        return cacheFile;
+    }
 
-        Path cacheFile = MCRUtils.safeResolve(idCachePath, baseId);
-        if (!Files.exists(cacheFile)) {
+    private static Path getCacheDirPath() {
+        Path dataDir = getDataDirPath();
+
+        Path idCachePath = dataDir.resolve("id_cache");
+        if (!Files.exists(idCachePath)) {
```

Comment:
Same here, avoid nesting.

Reply:
Fixed in https://github.com/MyCoRe-Org/mycore/pull/2095